crates/core/src/hash.rs

```diff
@@ -155,12 +154,11 @@ impl Hash {
     pub const fn as_ptr(&self) -> *const u8 {
         self.0.as_ptr()
     }
-}
-
-impl<'a> Add<&'a Hash> for Hash {
-    type Output = Self;
 
-    fn add(self, rhs: &'a Hash) -> Self::Output {
+    /// Given hashes A and B, compute Sha256(A||B),
+    /// but if one value is the zero hash, the other
+    /// value is returned.
+    pub fn concat(self, rhs: &Hash) -> Self {
         let mut hasher = Sha256::default();
```

Comment:
maybe we can defer the construction of the hash state to the if-branch where it is actually used